library/systemd/src/lib/yast2/system_service.rb

```diff
@@ -0,0 +1,490 @@
+    # @see #active=
+    #
+    # @return [Symbol] :start
+    def start
+      self.active = true
+      self.action = :start
+    end
+
+    # Sets the service to be stopped after calling {#save}
+    #
+    # @see #active=
+    #
+    # @return [Symbol] :stop
+    def stop
+      self.active = false
+      self.action = :stop
+    end
+
+    # Sets the service to be restarted after calling {#save}
+    #
+    # @return [Symbol] :restart
+    def restart
+      register_change(:active, true)
+      self.action = :restart
+    end
+
+    # Sets the service to be reloaded after calling {#save}
+    #
+    # @return [Symbol] :reload
+    def reload
+      register_change(:active, true)
+      self.action = :reload
+    end
+
+    # Saves changes into the underlying system
+    #
+    # @note Cached changes are reset and the underlying service is refreshed.
+    #
+    # @param keep_state [Boolean] Do not change service status. Useful when running on 1st stage.
+    #
+    # @return [Boolean] true if the service was saved correctly; false otherwise.
+    def save(keep_state: false)
+      clear_errors
+      save_start_mode
+      perform_action unless keep_state
+      reset && refresh
+    end
+
+    # Reverts cached changes
+    #
+    # @return [Boolean] true if the service was reset correctly. Actually, the
+    #   service always can be reset.
+    def reset
+      clear_changes
+      @action = nil
+
+      true
+    end
+
+    # Refreshes the underlying service
+    #
+    # @return [Boolean] true if the service was refreshed correctly; false otherwise.
+    def refresh
+      service.refresh!
+      @start_modes = nil
+      @current_start_mode = nil
+      true
+    rescue Yast::SystemctlError
+      false
+    end
+
+    # Whether there is any cached change that will be applied by calling {#save}.
+    #
+    # Some specific change can be checked by using the key parameter.
+    #
+    # @example
+    #   service.changed?(:start_mode)
+    #
+    # @return [Boolean]
+    def changed?(key = nil)
+      key ? changes.key?(key) : changes.any?
+    end
+
+  private
+
+    # @!method action=(value)
+    #
+    # Action to perform when the service is saved (see {#save})
+    #
+    # @param value [Symbol] :start, :stop, :restart, :reload
+    attr_writer :action
+
+    # @return [Hash<Symbol, Object>]
+    attr_reader :changes
+
+    # Sets whether the service should be active or not
+    #
+    # The given value will be applied after calling {#save}.
+    #
+    # @param value [Boolean] true to set this service as active
+    def active=(value)
+      if value == current_active?
+        unregister_change(:active)
+      else
+        register_change(:active, value)
+      end
+    end
+
+    # Sets start mode to the underlying system
+    def save_start_mode
+      return unless changes[:start_mode]
+
+      result =
+        case changes[:start_mode]
+        when :on_boot
+          service.enable && (socket ? socket.disable : true)
```

Comment:
SystemdUnit currently does not document the return value of #enable and #disable, please fix that.